### PR TITLE
feat(shared-dimensions): support DELETE

### PR DESCRIPTION
**Pull request by tpluscode**

### CODE REVIEW

File: apis/managed-dimensions/hydra/index.ttl [hunk 1]
```diff
@@ -49,6 +50,28 @@ md:ManagedDimensions
...
+             a code:EcmaScript ;
+             code:link <file:handlers/resource#DELETE>
+           ] .
+
+schema:DefinedTermSet
+  a hydra:Class ;
+  query:cascadeDelete
+    [
+      sh:inversePath schema:inDefinedTermSet ;
+    ] ;
```
Comment (tpluscode) [inline apis/managed-dimensions/hydra/index.ttl:66]: This SHACL Path is converted into a SPARQL query path to find and delete terms contained by a term set

File: apis/managed-dimensions/lib/domain/resource.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  await store.delete(term)
+
+  for (const cascadeDeleted of resourcesToDelete) {
+    cascadeDelete({
+      term: cascadeDeleted,
+      store,
+      api,
+    }).catch(e => {
+      error('Could not delete %s: %s', term.value, e.message)
+    })
```
Comment (tpluscode) [inline apis/managed-dimensions/lib/domain/resource.ts:70]: Any cascading removals are handled without blocking the request and errors are only recorded

File: apis/managed-dimensions/lib/domain/resource.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  if (path.term.termType === 'NamedNode') {
+    return path.term
+  }
+
+  if (path.term.termType !== 'BlankNode') {
+    throw new Error('Unexpected SHACL path')
+  }
+
+  if (path.has(sh.inversePath).term) {
+    return sparql`^${path.out(sh.inversePath).term}`
```
Comment (tpluscode) [inline apis/managed-dimensions/lib/domain/resource.ts:31]: Only support `sh:inversePath` for now

File: apis/managed-dimensions/hydra/index.ttl [hunk 4]
```diff
@@ -50,6 +50,28 @@ md:ManagedDimensions
        ]
 .
 
+_:DELETE a hydra:Operation, schema:DeleteAction ;
+         hydra:method "DELETE" ;
```
Comment (martinmaillard) [inline apis/managed-dimensions/hydra/index.ttl:54]: `hydra:title` would be nice

**Comment by martinmaillard** [comment]
Deleting a term seems to work, but the network request to the API never returns a response.

File: apis/managed-dimensions/lib/handlers/resource.ts [hunk 5]
```diff
@@ -1,8 +1,15 @@
...
-  const resource = await store().load(req.hydra.resource.term)
+export const DELETE = asyncMiddleware(async (req, res) => {
+  await cascadeDelete({
+    store: store(),
+    term: req.hydra.resource.term,
+    api: clownface(req.hydra.api),
+  })
 
-  return res.dataset(resource.dataset)
+  res.status(NO_CONTENT)
```
Comment (martinmaillard) [inline apis/managed-dimensions/lib/handlers/resource.ts:14]:
```suggestion
  res.sendStatus(NO_CONTENT)
```

**Review by martinmaillard — APPROVED** [review]

**Pull request merged**